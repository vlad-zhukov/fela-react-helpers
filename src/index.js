import {createElement} from 'react'; // eslint-disable-line
import PropTypes from 'prop-types';
import {createComponentFactory} from 'fela';

const propTypes = {
    renderer: PropTypes.object,
    theme: PropTypes.object,
};

const factory = createComponentFactory(createElement, propTypes);

function passThroughProps(props) {
    return Object.keys(props);
}

// eslint-disable-next-line import/prefer-default-export
export function withStyle(rule) {
    return Component => factory(rule, Component, passThroughProps);
}
