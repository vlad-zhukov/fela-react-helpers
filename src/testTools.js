import {renderToString} from 'fela-tools';
import {css} from 'js-beautify';

const defaultOptions = {
    indent_size: 2,
    end_with_newline: true,
    selector_separator_newline: false,
    newline_between_rules: false,
};

// eslint-disable-next-line import/prefer-default-export
export function renderToPrettyString(renderer, options = {}) {
    const styles = renderToString(renderer);
    return `\n${css(styles, Object.assign({}, defaultOptions, options))}`;
}
