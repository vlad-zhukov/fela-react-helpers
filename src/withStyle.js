import {createElement} from 'react'; // eslint-disable-line
import PropTypes from 'prop-types';
import {createComponentFactory} from 'fela-bindings';
import {withTheme} from 'react-fela';

const propTypes = {
    renderer: PropTypes.object,
    theme: PropTypes.object,
};

const factory = createComponentFactory(createElement, withTheme, propTypes);

export function passThroughProps(props) {
    const keys = Object.keys(props);
    const index = keys.indexOf('innerRef');
    if (index !== -1) {
        keys.splice(index, 1);
    }
    return keys;
}

export default function withStyle(rule) {
    return Component => factory(rule, Component, passThroughProps);
}
