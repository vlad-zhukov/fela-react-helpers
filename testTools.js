module.exports = require('./dist/fela-react-helpers.testTools');
