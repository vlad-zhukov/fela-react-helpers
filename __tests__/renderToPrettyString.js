import {createRenderer} from 'fela';
import {renderToPrettyString} from '../src/testTools';

function felaRenderer() {
    const fela = createRenderer();
    fela.renderStatic({boxSizing: 'border-box', margin: 0}, 'html,body');
    fela.renderStatic({fontSize: '14px'}, 'p');
    return fela;
}

it('should prettify styles', () => {
    const renderer = felaRenderer();
    expect(renderToPrettyString(renderer)).toMatchSnapshot();
});
