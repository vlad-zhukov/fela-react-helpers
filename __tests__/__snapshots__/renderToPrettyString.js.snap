// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should prettify styles 1`] = `
"
html, body {
  box-sizing: border-box;
  margin: 0
}
p {
  font-size: 14px
}
"
`;
